using System;

public class TrycatchExp
{
    public static void Main(string[] args)
    {
        try{
            Console.WriteLine("Enter no1");
            int x= int.Parse(Console.ReadLine());
            Console.WriteLine("Enter no2");
            int y= int.Parse(Console.ReadLine());
            int z=x/y;
        }
        catch(DivideByZeroException ex1)
        {
            Console.WriteLine(ex1.Message);
        }
        catch(FormatException ex2)
        {
            Console.WriteLine(ex2.Message);
        }
        catch(Exception ex)
        {
            Console.WriteLine(ex.Message);
            Console.WriteLine("Tata Bye Bye");
        }
        Console.WriteLine("End of the Program");
    }

}
//op
Enter no1
100
Enter no2
777777777777777777777777777777777
Value was either too large or too small for an Int32.
Tata Bye Bye
End of the Program

//op
Enter no1
100
Enter no2
y
Input string was not in a correct format.
End of the Program

//op
Enter no1
100
Enter no2
0
Attempted to divide by zero.
End of the Program

//op
Enter no1
24
Enter no2
12
End of the Program

.............................................................................................................................

using System;

public class TrycatchExp
{
    public static void Main(string[] args)
    {
            Console.WriteLine("Enter no1");
            int x= int.Parse(Console.ReadLine());
            Console.WriteLine("Enter no2");
            int y= int.Parse(Console.ReadLine());
            
            if(y%2 > 0)
            {
                throw new ApplicationException("Divisor value can't be odd!!");
            }
            int z=x/y;
        Console.WriteLine($"Result : {z}");
        Console.WriteLine("End of the Program");
    }
}
//op
Enter no1
100
Enter no2
21
Unhandled Exception:
System.ApplicationException: Divisor value can't be odd!!
  at TrycatchExp.Main (System.String[] args) [0x00038] in <76d49f88c8d549df818969e9a34c4aaf>:0
[ERROR] FATAL UNHANDLED EXCEPTION: System.ApplicationException: Divisor value can't be odd!!
  at TrycatchExp.Main (System.String[] args) [0x00038] in <76d49f88c8d549df818969e9a34c4aaf>:0
.....................................................................................................
using System;

public class TrycatchExp
{
    public static void Main(string[] args)
    {
            Console.WriteLine("Enter no1");
            int x= int.Parse(Console.ReadLine());
            Console.WriteLine("Enter no2");
            int y= int.Parse(Console.ReadLine());
            
            if(y%2 > 0)
            {
                throw new ApplicationException("Divisor value can't be odd!!");
            }
            int z=x/y;
        Console.WriteLine($"Result : {z}");
        Console.WriteLine("End of the Program");
    }
}

//op
Enter no1
100
Enter no2
0
Unhandled Exception:
System.DivideByZeroException: Attempted to divide by zero.
  at TrycatchExp.Main (System.String[] args) [0x00043] in <196dcf4aefbc46878d8188303118ed03>:0 
[ERROR] FATAL UNHANDLED EXCEPTION: System.DivideByZeroException: Attempted to divide by zero.
  at TrycatchExp.Main (System.String[] args) [0x00043] in <196dcf4aefbc46878d8188303118ed03>:0 
.....................................................................................................

using System;

public class TrycatchExp
{
    public static void Main(string[] args)
    {
        try{
            Console.WriteLine("Enter no1");
            int x= int.Parse(Console.ReadLine());
            Console.WriteLine("Enter no2");
            int y= int.Parse(Console.ReadLine());
            int z=x/y;
            Console.WriteLine($"The result is: {z}");
        }
        catch(Exception ex)
        {
            Console.WriteLine(ex.Message);
            Console.WriteLine("Tata Bye Bye");
        }
        finally
        {
            Console.WriteLine("Finally End of the Program");
        }
    }

}
//op
Enter no1
500
Enter no2
25
5
The result is: 20
Finally End of the Program

//op
Enter no1
500
Enter no2
m
Input string was not in a correct format.
Tata Bye Bye
Finally End of the Program

//op
Enter no1
200
Enter no2
777777777777777777777777777777
Value was either too large or too small for an Int32.
Tata Bye Bye
Finally End of the Program

......................................................................................................
using System;
public class TrycatchExp
{
    public static void Main(string[] args)
    {
        try{
            Console.WriteLine("Enter no1");
            int x= int.Parse(Console.ReadLine());
            Console.WriteLine("Enter no2");
            int y= int.Parse(Console.ReadLine());
            if(y==1)
            {
                return;
            }
            int z=x/y;
            Console.WriteLine($"The result is: {z}");
        }
        catch(Exception ex)
        {
            Console.WriteLine(ex.Message);
            Console.WriteLine("Tata Bye Bye");
        }
        finally
        {
            Console.WriteLine("Finally End of the Program");
        }
        Console.WriteLine("DONE");
    }
}

//op
Enter no1
100
Enter no2
1
Finally End of the Program
............................................................................................
using System;
public class DivisorCannotBeOdd : ApplicationException
{
    public override string Message
    {
        get
        {
            return "Divisor value can't be odd!!";
            
        }
    }
}
public class TrycatchExp
{
    public static void Main(string[] args)
    {
            Console.WriteLine("Enter no1");
            int x= int.Parse(Console.ReadLine());
            Console.WriteLine("Enter no2");
            int y= int.Parse(Console.ReadLine());
            
            if(y%2 > 0)
            {
                throw new DivisorCannotBeOdd();
            }
            int z=x/y;
        Console.WriteLine($"Result : {z}");
        Console.WriteLine("End of the Program");
    }
}

//op
Enter no1
154
Enter no2
13
Unhandled Exception:
DivisorCannotBeOdd: Divisor value can't be odd!!
  at TrycatchExp.Main (System.String[] args) [0x0003d] in <bc994dd822694f709ea7f4a752b04012>:0 
[ERROR] FATAL UNHANDLED EXCEPTION: DivisorCannotBeOdd: Divisor value can't be odd!!
  at TrycatchExp.Main (System.String[] args) [0x0003d] in <bc994dd822694f709ea7f4a752b04012>:0 
..........................................................................................................
using System;
using System.IO;
using System.Runtime.Serialization;
public class UserInfoException : Exception
{
    public UserInfoException():base()
    {}
    public UserInfoException(string message) : base(message)
    {}
    public UserInfoException(string message, Exception innerException) : base(message,innerException)
    {}
    public UserInfoException(SerializationInfo info,StreamingContext context) : base(info,context)
    {}
}
public class TrycatchExp
{
    public static void Main(string[] args)
    {
        try{
            throw new UserInfoException("No duplicates allowed");
        }catch(UserInfoException ex)
        {
            Console.WriteLine(ex.Message);
        }
    }
}

//op
No duplicates allowed

